src/components/implementation/fork_tests/voter/voter.c

```diff
@@ -89,17 +92,20 @@ int nwrite(spdid_t spdid, replica_type to, size_t sz) {
 		return -1; // whatever, shouldn't happen if blocked
 	}
 	else {
+		assert(replica->buf_write);
+		assert(sz <= 1024);
+		
 		replica->destination = to_comp;
+		replica->sz_write = sz;
 		printc("resplica destination set to %x\n", replica->destination);
-		assert(replica->buf_write);
 		ret = block_replica(replica);
 		if (ret < 0) printc("Error\n");
 		/* On wakeup */	
 		return 0;
 	}
 }
 
-int nread(spdid_t spdid, replica_type from, size_t sz) {
+size_t nread(spdid_t spdid, replica_type from, size_t sz) {
```

Comment:
Okay, I think we are on the same page for it. The voter will not read more that sz bytes. As of right now, sz isn't being used though.